index.js

```diff
@@ -316,6 +316,10 @@ function HttpRequest(options) {
 			}
 		};
 	}
+
+	if ('onerror' in xhr && !xhr.onreadystatechange) {
+		xhr.onerror = onLoad;
```

Comment:
I would like to, but I'm not sure if `'onreadystatechange' in xhr` test works in IE. I remember the behavior being quite peculiar. Probably a real b*tch to try out now.